lib/LTP/WhiteList.pm

```diff
@@ -19,23 +18,28 @@ use Mojo::UserAgent;
 use Mojo::File 'path';
 use YAML::PP;
 
-our @EXPORT = qw(
-  download_whitelist
-  find_whitelist_testsuite
-  find_whitelist_entry
-  override_known_failures
-  is_test_disabled
-  list_skipped_tests
-);
+sub new {
+    my $class = shift;
+    my $self = bless({}, $class);
+    my $path = get_var('LTP_KNOWN_ISSUES_LOCAL');
+
+    if (!defined($path) && get_var('LTP_KNOWN_ISSUES')) {
+        $path = _download_whitelist();
+    }
 
-sub download_whitelist {
+    $self->{whitelist} = _load_whitelist_file($path) if $path;
+    $self->{whitelist} ||= {};
+    return $self;
+}
+
+sub _download_whitelist {
     my $path = get_var('LTP_KNOWN_ISSUES');
     return undef unless defined($path);
 
     my $res = Mojo::UserAgent->new->get($path)->result;
     unless ($res->is_success) {
         record_info("File not downloaded!", $res->message, result => 'softfail');
-        set_var('LTP_KNOWN_ISSUES_LOCAL', undef);
+        set_var('LTP_KNOWN_ISSUES_LOCAL', '');
```

Comment:
i dont think you need this statement anymore. Where is it used?

Reply:
It's used in `LTP::WhiteList::new()`:
- `undef` => download the whitelist file
- empty string => do nothing
- non-empty value => just load the already downloaded file